git-config/src/values.rs

```diff
@@ -237,6 +243,7 @@ impl From<&Value<'_>> for Vec<u8> {
             Value::Boolean(b) => b.into(),
             Value::Integer(i) => i.into(),
             Value::Color(c) => c.into(),
+            Value::Path(p) => p.value.to_str().expect("path is valid utf-8").as_bytes().into(),
```

Comment:
No encoding can be assumed, definitely not UTF-8. I didn't look into git's source code yet to see what kind of requirements apply to a path stored in a git-config file. The assure round-tripping, what's stored in `p` can't be the decoded value. This crate is already provide means of 'normalizing' values, which deals with escaping values and the likes, probably using this machinery be valuable here.

Reply:
Yep, I also feel like looking into git's source code to see how they handle path. And will also look closer at 'normalizing' values.

But see also how existing code "assumes" UTF-8. For example, 
       ```
 T::try_from(self.get_raw_value(section_name, subsection_name, key)?)
            .map_err(|_| GitConfigError::FailedConversion)
```
would go to 
`impl TryFrom<Cow<'_, [u8]>> for Integer {`
then to e.g
`Self::from_str(std::str::from_utf8(s).map_err(|_| ())?).map_err(|_| ())`

So we can get `GitConfigError::FailedConversion`

That should be ok either?

Reply:
If it was enforced, it indeed wouldn't. However, here it seems the caller decides which type to use and if that is `String` or `str`, they probably know the value is not expected to be encoded differently.

The existing usages of `str::from_utf8(…)` also only appear in places where this is expected to succeed, i.e. for decimals or booleans, which are known to be ascii only.